orcestra_book/plans/HALO-20240903a.md

```diff
@@ -0,0 +1,318 @@
+---
+arrival_airport: GVAC
+categories: [ec_under, ec_track, c_north, c_mid, c_south, c_atr]
+crew:
+- job: PI
+  name: "Silke Gro\xDF"
+- job: WALES
+  name: Julia Windmiller
+- job: HAMP
+  name: Jakob Deutloff
+- job: Dropsondes
+  name: Theresa Mieslinger
+- job: Smart/VELOX
+  name: TBD
+- job: SpecMACS
+  name: TBD
+- job: Flight Documentation
+  name: TBD
+- job: Ground contact
+  name: TBD
+departure_airport: GVAC
+flight_id: HALO-20240831a
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+landing: '2024-08-31 17:44:00Z'
+orphan: true
+platform: HALO
+takeoff: '2024-08-31 10:11:00Z'
+---
+
+{logo}`PERCUSION`
+
+# Flight plan - {front}`flight_id`
+
+```{badges}
+```
+
+## Crew
+
+The flight is planned to take off at {front}`takeoff`.
+
+```{crew}
+```
+
+## Flight plan
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+from dataclasses import asdict
+from datetime import datetime
+import cartopy.crs as ccrs
+import easygems.healpix as egh
+import intake
+import matplotlib.pyplot as plt
+import numpy as np
+import orcestra
+import orcestra.flightplan as fp
+import orcestra.sat
+from orcestra.flightplan import LatLon, IntoCircle, bco, sal, mindelo, find_ec_lon, vertical_preview, to_kml\
+
+def ec_time_at_lat(ec_track, lat):
+    e = np.datetime64("2024-08-01")
+    s = np.timedelta64(1, "ns")
+    return (((ec_track.swap_dims({"time":"lat"}).time - e) / s).interp(lat=lat) * s + e)
+
+# Global coordinates and definitions that should not change from flight to flight
+
+lon_min, lon_max, lat_min, lat_max = -65, -5, -5, 25
+
+radius = 72e3*1.852
+atr_radius = 38e3*1.852
+
+band = "east"
+airport = sal if band == "east" else bco
+natal = LatLon(-5 - 47/60. - 42.00/3600.,-35 - 12/60. - 33.98/3600., label = "natal")
+
+# Define dates for forecast initialization and flight
+
+issued_time = datetime(2024, 8, 27, 0, 0, 0)
+
+flight_time = datetime(2024, 9, 3, 12, 0, 0)
+flight_index = f"HALO-{flight_time.strftime('%Y%m%d')}a"
+
+# adjust takeoff time to match EC overpass
+takeoff_time = np.datetime64("2024-09-03T10:21:32")
+
```

Comment:
This should not be defined, if the time of the earthcare meeting point is defined.
```suggestion
```

Reply:
I picked the latitude of the EC overpass and adjust the time.. but maybe there is a better way?

Reply:
If you assign the overpass time to the overpass waypoint, the flightplanning tool should figure out the takeoff time by itself.